sample/NodeRetry/HttpTrigger-RetryFunctionJson/index.js

```diff
@@ -2,9 +2,9 @@
 var errorString = 'An error occurred';
 
 module.exports = async function (context, req) {
-    if (context.executionContext.retryContext && (context.executionContext.retryContext.retryCount !== invocationCount
-        || !(context.executionContext.retryContext.maxRetryCount === 4 || context.executionContext.retryContext.maxRetryCount === 0)
-        || !(context.executionContext.retryContext.exception.message.includes(errorString)))) {
+    var retryContext = context.executionContext.retryContext;
+    if (retryContext.retryCount != invocationCount || retryContext.maxRetryCount != 4
```

Comment:
simplify. When writing the original test, we did not have retryContext:
```js
module.exports = async function (context, req) {
    context.log('JavaScript HTTP trigger function processed a equest.context.executionContext.retryContext.retryCount: ' + context.executionContext.retryContext.retryCount);
    if (context.executionContext.retryContext.retryCount < 4) {
            throw new Error(errorString);
        }
        context.res = {
            // status: 200, /* Defaults to 200 */
            body: "invocationCount: " + context.executionContext.retryContext.retryCount
        };
}
```

Reply:
I believe the checks I have are stronger - it ensures invocation count is correct, and retryCount is exactly as expected. E.g. your check would pass if retryCount were 100.

Reply:
at minimum please remove the else block that relies on static invocationcount - this was added for the very first version when there was not retry context.

Reply:
Done for both.

Reply:
Thank you!